### PR TITLE
Pastebin integration

### PR DESCRIPTION
**Pull request by yaleman**
Please accept this PR as a possible pack for Stackstorm.

### CODE REVIEW

File: st2-pastebin/sensors/pastebin_poller.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                    return
+                # sort by timestamp, it comes in most-recent-first
+                data = sorted(jsondata, key=lambda k: k['date'])
+                for paste in data:
+                    self._logger.debug("time:{} key:{}".format(paste['date'], paste['key']))
+                    if paste['date'] > self._get_last_time():
+                        # this is the timestamp of the last processed paste
+                        self._set_last_time(last_time=paste['date'])
+                        # do the thing
+                        payload = {'date' : int(paste['date']),
```
Comment (nmaludy) [inline st2-pastebin/sensors/pastebin_poller.py]: any reason you don't just include all of the fields from the response (curious)?

File: st2-pastebin/sensors/pastebin_poller.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            else:
+                self._logger.debug("No response from the API (status_code: {})".format(req.status_code))
+        except Exception as error_message:
+            self._logger.debug("Threw an error: {}".format(error_message))
+            self._logger.debug(traceback.format_exc())
+        return
+
+    def _get_last_time(self):
+        """ returns the last timestamp that was processed by the poller """
+        if not self._last_time and hasattr(self._sensor_service, 'get_value'):
```
Comment (nmaludy) [inline st2-pastebin/sensors/pastebin_poller.py]: these functions should always exist on the sensor_service, any reason you're checking for them?

File: st2-pastebin/actions/scrape_paste_metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+---
+name: "scrape_paste_metadata"
+runner_type: "local-shell-cmd"
```
[the remark this answers is not in the document]
Comment (yaleman) [reply st2-pastebin/actions/scrape_paste_metadata.yaml]: http request runner doesn't allow for setting the IPv4/IPv6 version - you can only whitelist one IP in the pastebin API - which breaks when your server has dual stack networking. Writing a whole extra python script seemed unnecessary.

File: st2-pastebin/sensors/pastebin_poller.py [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+import requests
+
+from st2reactor.sensor.base import PollingSensor
+
+__all__ = ['PasteBinPoller',]
+
+SCRAPE_URL = 'https://scrape.pastebin.com/api_scraping.php'
+
+old_getaddrinfo = socket.getaddrinfo
```
[the remark this answers is not in the document]
Comment (yaleman) [reply st2-pastebin/sensors/pastebin_poller.py]: It's the only way I could reliably force IPv4 vs IPv6 usage on `requests`, need line 15 to be able to revert it.

File: st2-pastebin/sensors/pastebin_poller.py (hunk 1 above)
Comment (yaleman) [reply st2-pastebin/sensors/pastebin_poller.py]: Because I needed to convert the formats.

**Comment by yaleman** [comment]
I can't use someone else's repo for my code, closing this.

**Pull request closed without merging**